### PR TITLE
Issue 229: Add explicit encoding, license and unix line separators.

### LINKED ISSUE #229: Failed build with python3.6
Hi,

   I failed build num2words-0.5.8 with python-3.6 on FreeBSD-11.2, while 2.7 and 3.7 success.
   The error message is : 
  =======================<phase: configure      >============================
===>   py36-num2words-0.5.8 depends on package: py36-setuptools>0 - found
===>   py36-num2words-0.5.8 depends on file: /usr/local/bin/python3.6 - found
===>   py36-num2words-0.5.8 depends on file: /usr/local/bin/ccache - found
===>  Configuring for py36-num2words-0.5.8
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "setup.py", line 20, in <module>
    LONG_DESC = open('README.rst', 'rt').read() + '\n\n' + \
  File "/usr/local/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 1557: ordinal not in range(128)
*** Error code 1

Any suggestions ?

wen

### PR DESCRIPTION
**Pull request by erozqba**
### Changes proposed in this pull request:

Some maintainers for different Linux distribution are having some
problems with the encoding in different python versions. This change
intents to make all python files define explicitly the encoding, license, 
and Unix line separators. Remove one example of the README.rst file
where a possible non UTF-8 character is used.

### Status

- [X] READY
- [ ] HOLD
- [ ] WIP (Work-In-Progress)

### How to verify this change

Make sure all tests pass and coverage remains the same.

### CODE REVIEW

**Comment by null-git** [comment]
Thanks for this, but it doesn't build either:
```
===>  Configuring for py36-num2words-0.5.8_1
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "setup.py", line 59, in <module>
    version=find_version("bin/num2words"),
  File "setup.py", line 47, in find_version
    for line in fp:
  File "/usr/local/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 747: ordinal not in range(128)
*** Error code 1
```
This is the same example in bin/num2words:27 that was removed from README.rst. Removing this line as well does the trick. But wouldn't it be nicer to just open those specific files as utf-8? As they are opened by open(), the "coding: utf-8" line doesn't do anything for this. Replacing open() with io.open() might do it? This seems to be available for python2.7 as well and to be the same as open() in python3+

**Comment by erozqba** [comment]
@null-git I agree! I will also use the `io.open(..., encoding="utf-8")`

File: setup.py [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+# MA 02110-1301 USA
+
 import re
+from io import open
```
Comment (erozqba) [inline setup.py]: @null-git I'm importing `open` from `io` so it should work on Python 2.7.

**Comment by null-git** [comment]
Sorry, I completely missed that. I think it's fine how you did it, other people won't care to much which open() this is and as python 2.7 is hopefully gone soonish anyway, both functions will be the same for most people anyway. Looks good to me.

File: bin/num2words [hunk 2]
```diff
@@ -1,5 +1,20 @@
 #!/usr/bin/env python
-# -*- coding: utf-8 -*-
```
Comment (pierreduchemin) [inline bin/num2words]: Why is this line removed here but added everywhere else?

File: num2words/lang_UK.py [hunk 3]
```diff
@@ -1,174 +1,175 @@
...
-                words.append(self.pluralize(x, THOUSANDS[i]))
-
-        return ' '.join(words)
-
-    def _cents_verbose(self, number, currency):
-        return self._int2word(number, currency == 'UAH')
-
-    def to_ordinal(self, number):
-        raise NotImplementedError()
+# -*- encoding: utf-8 -*-
```
Comment (pierreduchemin) [inline num2words/lang_UK.py]: Shouldn't it be "coding" instead of "encoding"?

File: num2words/utils.py [hunk 4]
```diff
@@ -1,17 +1,34 @@
...
-            result = n[i:i+x]
-            yield int(result) if format_int else result
-    else:
-        yield int(n) if format_int else n
-
-
-def get_digits(n):
-    a = [int(x) for x in reversed(list(('%03d' % n)[-3:]))]
-    return a
+# encoding: UTF-8
```
Comment (pierreduchemin) [inline num2words/utils.py]: Shouldn't it be "coding" instead of "encoding"?

File: num2words/lang_SL.py [hunk 5]
```diff
@@ -16,8 +16,6 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
 # MA 02110-1301 USA
 
-# -*- coding: utf-8 -*-
```
Comment (pierreduchemin) [inline num2words/lang_SL.py]: Why is this removed here but added everywhere else?

Comment (erozqba) [reply num2words/lang_SL.py]: It's already at the top of the file.

Comment (erozqba) [reply num2words/lang_SL.py]: I made a mistake during my rebase, I will fix it later.

File: setup.py (hunk 1 above)
Comment (pierreduchemin) [reply setup.py]: I don't think it's required and it's a bit outside of the scope of this MR.
Feel free to create another MR or change this before merge if you feel this is necessary :+1:

Comment (erozqba) [reply setup.py]: @pierreduchemin, In my opinion, it's not outside the scope of this PR, the actual issue is #229 and all these changes are trying to fix that. On the last release, we have included some examples with characters that are not UTF-8 on some text files that aren't `*.py` example the README.md. Using this to open these files as UTF-8 is a way to avoid the issue described in #229. I just decided that was a good idea to review all the headers and add the explicit encoding on the python files too.

**Comment by erozqba** [comment]
@pierreduchemin Please review this PR again, it should be ready. Please review carefully the files on which I have changed the line end. Thanks a lot.
